keras_cv/models/segmentation/deeplabv3plus.py

```diff
@@ -0,0 +1,319 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.layers.spatial_pyramid import SpatialPyramidPooling
+from keras_cv.models import utils
+from keras_cv.models.weights import parse_weights
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class DeepLabV3Plus(keras.Model):
+    """
+    A segmentation model based on the DeepLabV3Plus model.
+
+    Args:
+        classes: int, the number of classes for the detection model. Note that
+            the classes doesn't contain the background class, and the classes
+            from the data should be represented by integers with range
+            [0, classes).
+        backbone: a backbone for the model, expected to be a KerasCV model.
+            Typically ResNet50V2 or ResNet101V2. Default 'low_level_feature_layer' assumes
+            either and uses 'v2_stack_1_block4_1_relu' for them by default.
+        low_level_feature_layer: the layer name for the low-level features to use for encoding/decoding
+            spatial information for the supplied backbone. The high-level activations come from the last layer in the model.
+        weights: weights for the complete DeepLabV3Plus model. one of `None` (random
+            initialization), a pretrained weight file path, or a reference to
+            pre-trained weights (e.g. 'imagenet/classification') (see available
+            pre-trained weights in weights.py)
+        spatial_pyramid_pooling: also known as Atrous Spatial Pyramid Pooling (ASPP).
+            Performs spatial pooling on different spatial levels in the pyramid, with
+            dilation.
+        segmentation_head: an optional `tf.keras.Layer` that predict the segmentation
+            mask based on feature from backbone and feature from decoder.
+        segmentation_head_activation: default 'softmax', the activation layer to apply after
```

Comment:
activation should be part of segmentation head

Reply:
Whoops - updating. Thanks!

Reply:
Seems it's not updated yet?

Reply:
Please make activation as part of segmentation head